### PR TITLE
feat: enhance plugin Manager with visibility

### PR DESCRIPTION
**Pull request by haneabogdan**
This PR...

## Changes

-

## Fixes

-

## How to test it

-

## screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/pages/TestSuites/TestSuitesList/TestSuitesList.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ const TestSuitesList: FC = () => {
       onItemAbort={onItemAbort}
       entity="test-suites"
       pageTitle="Test Suites"
-      pageTitleAddon={usePluginSlot('testSuiteTitleIcon')}
+      pageTitleAddon={usePluginSlot('testSuiteTitleAddon')}
```
Comment (rangoo94) [inline src/components/pages/TestSuites/TestSuitesList/TestSuitesList.tsx:72]:
```suggestion
      pageTitleAddon={usePluginSlot('testSuitesListTitleAddon')}
```

File: src/plugins/types.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export interface Plugin {
 
 export interface SlotMetaData {
   order?: number;
+  visibile?: (params: any) => boolean;
```
Comment (rangoo94) [inline src/plugins/types.ts:13]:
```suggestion
  visible?: (params: any) => boolean;
```

Besides, do we need `params` there? These are declare in the `setup` function, which already has access to the `scope`:
```suggestion
  visible?: () => boolean;
```

File: src/plugins/pluginHooks.ts [hunk 3]
```diff
@@ -14,6 +14,20 @@ export const usePluginSlot = (name: string) => usePluginSlotList(name)[0];
...
+      // unless visibility is configured explicitly in the metadata
+      if (!e.metadata?.visibile) {
+        defaults.push(e);
+      } else if (e.metadata.visibile(scope.getState(name))) {
+        // if visibility is configured, check if the element is visible
+        defaults.push(e);
+      }
+    });
+    return orderArray(defaults);
+  }, [defaults, scope.getSlot(name)]);
```
Comment (rangoo94) [inline src/plugins/pluginHooks.ts:31]: Could be a single `if` condition (`!e.metadata?.visible || e.metadata.visible()`).

Besides, you may use `Array.prototype.filter`:

```ts
  // by default, all elements are visible,
  // unless visibility is configured explicitly in the metadata
  const elements = useMemo(() => orderArray([
    ...defaults,
    ...scope.getSlot(name).filter(x => (!x.metadata?.visible || x.metadata.visible())),
  ]), [defaults, scope.getSlot(name)]);
```

Additionally, you could consider filtering `defaults` for consistency too:

```ts
  // by default, all elements are visible,
  // unless visibility is configured explicitly in the metadata
  const elements = useMemo(() => orderArray(
    [...defaults, ...scope.getSlot(name)]
      .filter(x => (!x.metadata?.visible || x.metadata.visible()))
  ), [defaults, scope.getSlot(name)]);
```

**Comment by rangoo94** [comment]
We could use `visibility` already for `defaultExecutionTabs` - we are doing filtering there, while these `metadata` would work 👍

File: src/components/molecules/MessagePanel/MessagePanel.tsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ import {
 interface ButtonConfig {
```
[the remark this answers is not in the document]
Comment (haneabogdan) [reply src/components/molecules/MessagePanel/MessagePanel.tsx:15]: `ICustomButtonProps` already extends `ButtonProps` from `antd`. will switch to `ICustomButtonProps`

File: src/components/molecules/ExecutionDetails/TestExecutionDetails/TestExecutionDetailsTabs.tsx [hunk 5]
```diff
@@ -132,18 +130,17 @@ const TestExecutionDetailsTabs: React.FC = () => {
...
+        key: 'Variables',
+        label: 'Variables',
+        children: <ExecutionsVariablesList variables={decomposedVars} />,
+      },
+      metadata: {
+        order: 1,
+        visible: () => decomposedVars.length,
+      },
+    },
   ].filter(Boolean);
```
Comment (rangoo94) [inline src/components/molecules/ExecutionDetails/TestExecutionDetails/TestExecutionDetailsTabs.tsx:144]: `filter` is no longer needed

**Review by rangoo94 — APPROVED** [review]

**Pull request merged**